client/components/data/query-simple-payments/index.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestProduct, requestProducts } from 'state/simple-payments/product-list/actions';
+
+class QuerySimplePayments extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			nextProps.siteId !== this.props.siteId ||
+			nextProps.productId !== this.props.productId
+		) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		const { siteId } = props;
+
+		if ( ! props.siteId ) {
```

Comment:
Use the `siteId` variable you just destructured on the previous line :) My style preference would be to either destructure either all needed props, or none of them.